### PR TITLE
[Feature Request] github milestone resource and data source

### LINKED ISSUE #295: [Feature] Add milestone support
## Overview
I want to manage the Repository Milestones with Terraform.

## Proposed new resources

This functionality would probably require two new resources/data sources at a minimum.

    New resource: github_repository_milestone
    New data source: github_repository_milestone

## Possible configuration

```tf
resource "github_repository_milestone" "my_milestone" {
  name         = "My Milestone"
  repository   = "test-repo"
  description  = "some info" // optional
  due_date     = "yyyy-mm-dd" // optional
}
```

## Related materials

- https://github.com/features/project-management/

### PR DESCRIPTION
**Pull request by anGie44**
Closes #295 

Changes include:
* new resource/data-source: `github_repository_milestone`
* import support with: `owner/repository/number`

Output of acceptance tests:
```
--- PASS: TestAccGithubRepositoryMilestone_disappears (13.65s)
--- PASS: TestAccGithubRepositoryMilestone_update (17.49s)
--- PASS: TestAccGithubRepositoryMilestone_basic (17.65s)
--- PASS: TestAccGithubRepositoryMilestone_multiple (21.82s)

--- PASS: TestAccGithubRepositoryMilestoneDataSource_noMatchReturnsError (0.34s)
--- PASS: TestAccGithubRepositoryMilestoneDataSource_existing (8.96s)
```

### CODE REVIEW

File: github/resource_github_repository_milestone.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+	}
+}
+
+const (
+	layoutISO = "2006-01-02"
+)
+
+func resourceGithubRepositoryMilestoneCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*Organization).v3client
+	ctx := context.Background()
```
Comment (jcudit) [inline github/resource_github_repository_milestone.go:85]: Recently came across this:

https://github.com/terraform-providers/terraform-provider-github/blob/518829b4f87f8613feaa86b4046702a962619830/github/provider.go#L136

Does the provider become more responsive to a cancel operation if we wire this context all the way through to requests like this?

**Comment by ewilde** [comment]
@jcudit is there anything left to do on this PR. I'm happy to help out if it's missing anything :+1:

**Comment by jcudit** [comment]
@ewilde thanks for the ping on this one 😄 

This has been rebased and the tests have been brought in line with the latest suite.  Look out for an upcoming CHANGELOG PR and raise any issues once this releases.  Happy to work with any data you can bring back on the next bugfix release if this doesn't play well with your use case.

**Comment by ewilde** [comment]
@jcudit fantastic, thank you for taking the time to get this feature into the provider :+1: It looks like it will be a good match for my basic use case

**Pull request merged**